src/scripts/roll.ts

```diff
@@ -13,7 +13,14 @@ import { Robot } from "tsbot";
 let roll = (robot: Robot) => {
     robot.respond(/roll( me)?( (\d+))?/i, (res) => {
         let max = Math.abs(Number(res.match[2])) || 100;
-        res.reply("rolls a " + Math.floor(Math.random() * max + 1) + "!");
+        let roll = Math.floor(Math.random() * max + 1);
+
+        if (max == 100) {
+            robot.emit("sniperino:roll", res, roll);
```

Comment:
not sure the event should be "sniperino:roll" you are essentially coupling roll script with the sniperino script. In my mind, the roll script is the roll script, it has no knowledge of the sniperino script it does the following:

Rolls out of X (X provided by the input as you have here) and also fires an event every time as well as displaying the output. The event could contain the maximum roll and the roll itself. Sniperino script could then listen to any "roll" event which provides a roll and the maximum and decide what to do.